src/helpers/helpers.ts

```diff
@@ -96,17 +97,19 @@ enum authType {
  * @param {CredentialType} credential - The CredentialType object representing authentication information.
  * @returns {string} - A string indicating the authentication type, e.g., "Username and Password".
  */
-const getAuthType = ({ username, password, auth_token, ssh_key }: CredentialType): authType => {
-  if (username && password) {
-    return authType.UsernameAndPassword;
+const getAuthType = ({ auth_type }: CredentialType): authType => {
+  switch (auth_type) {
+    case 'password':
+      return authType.UsernameAndPassword;
+    case 'auth_token':
+      return authType.Token;
+    case 'ssh_key':
+      return authType.SSHKey;
+    case 'ssh_keyfile':
+      return authType.SSHKeyFile;
+    default:
+      throw new Error('Unknown credential type');
```

Comment:
Will this error be handled anywhere? If not, change the behavior here.

This is usually an OK behavior for server side, but not for UI - especially when you are NOT dealing with user input.

Reply:
this was existing... if it wasn't called out before I vote for iterating on it

Reply:
looking back through the slack thread Brad posted the error, yes the throw is the issue...

Reply:
I'm happy with the new code. I'll leave unresolved just in case @cdcabrera wants to add anything else, otherwise feel free to solve this thread.